stress: flaky test (skipped) in storage: TestNodeLiveness

Binary: cockroach/static-tests.tar.gz sha: https://github.com/cockroachdb/cockroach/commits/9ef38d72441225bc3c665af3ae4a2b74d64d93d5

Stress build found a failed test:

```
=== RUN   TestNodeLiveness
I161014 05:16:24.869722 32484 storage/engine/rocksdb.go:349  opening in memory rocksdb instance
I161014 05:16:24.870853 32484 gossip/gossip.go:234  initial resolvers: []
W161014 05:16:24.870883 32484 gossip/gossip.go:1045  no resolvers found; use --join to specify a connected node
I161014 05:16:24.870947 32484 gossip/gossip.go:268  NodeID set to 1
I161014 05:16:24.874525 32484 gossip/gossip.go:268  NodeID set to 1
I161014 05:16:24.875173 32484 storage/store.go:1075  [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I161014 05:16:24.875252 32484 gossip/gossip.go:282  NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:44063" > attrs:<> locality:<> 
I161014 05:16:24.876014 32518 storage/replica_trigger.go:258  [s1,r1:/M{in-ax},r1:/M{in-ax}] new range lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 900ms following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0s [physicalTime=1970-01-01 00:00:00 +0000 UTC]
I161014 05:16:24.876614 32484 storage/engine/rocksdb.go:349  opening in memory rocksdb instance
I161014 05:16:24.877720 32484 gossip/gossip.go:234  initial resolvers: [127.0.0.1:44063]
W161014 05:16:24.877757 32484 gossip/gossip.go:1047  no incoming or outgoing connections
I161014 05:16:24.877813 32484 gossip/gossip.go:268  NodeID set to 2
I161014 05:16:24.879059 32484 gossip/gossip.go:268  NodeID set to 2
I161014 05:16:24.879260 32484 storage/store.go:1075  [n2,s2]: failed initial metrics computation: [n2,s2]: system config not yet available
I161014 05:16:24.879338 32484 gossip/gossip.go:282  NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:36382" > attrs:<> locality:<> 
I161014 05:16:24.879586 32484 storage/engine/rocksdb.go:349  opening in memory rocksdb instance
I161014 05:16:24.879642 32567 gossip/client.go:125  node 2: started gossip client to 127.0.0.1:44063
I161014 05:16:24.880580 32484 gossip/gossip.go:234  initial resolvers: [127.0.0.1:44063]
W161014 05:16:24.880608 32484 gossip/gossip.go:1047  no incoming or outgoing connections
I161014 05:16:24.880646 32484 gossip/gossip.go:268  NodeID set to 3
I161014 05:16:24.881761 32484 gossip/gossip.go:268  NodeID set to 3
I161014 05:16:24.881943 32484 storage/store.go:1075  [n3,s3]: failed initial metrics computation: [n3,s3]: system config not yet available
I161014 05:16:24.882010 32484 gossip/gossip.go:282  NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:38517" > attrs:<> locality:<> 
I161014 05:16:24.882855 32659 gossip/client.go:125  node 3: started gossip client to 127.0.0.1:44063
I161014 05:16:24.888870 30935 util/stop/stopper.go:353  stop has been called, stopping or quiescing all running tasks
I161014 05:16:24.889037 30935 util/stop/stopper.go:353  stop has been called, stopping or quiescing all running tasks
I161014 05:16:24.889198 30935 util/stop/stopper.go:353  stop has been called, stopping or quiescing all running tasks
I161014 05:16:24.890967 30935 util/stop/stopper.go:353  stop has been called, stopping or quiescing all running tasks
I161014 05:16:24.891307 32469 http2_client.go:1053  transport: http2Client.notifyError got notified that the client transport was broken EOF.
I161014 05:16:24.891387 30935 util/stop/stopper.go:353  stop has been called, stopping or quiescing all running tasks
I161014 05:16:24.891759 30935 util/stop/stopper.go:353  stop has been called, stopping or quiescing all running tasks
I161014 05:16:24.891883 30935 util/stop/stopper.go:353  stop has been called, stopping or quiescing all running tasks
--- FAIL: TestNodeLiveness (0.02s)
    node_liveness_test.go:93: expected metrics count == 2; got 1
```

Run Details:

```
0 runs so far, 0 failures, over 5s
0 runs so far, 0 failures, over 10s
0 runs so far, 0 failures, over 15s
0 runs so far, 0 failures, over 20s
0 runs so far, 0 failures, over 25s
0 runs so far, 0 failures, over 30s
15 runs so far, 0 failures, over 35s
16 runs so far, 0 failures, over 40s
16 runs so far, 0 failures, over 45s
16 runs so far, 0 failures, over 50s
16 runs so far, 0 failures, over 55s
16 runs so far, 0 failures, over 1m0s
29 runs so far, 0 failures, over 1m5s
32 runs so far, 0 failures, over 1m10s
32 runs so far, 0 failures, over 1m15s
32 runs so far, 0 failures, over 1m20s
32 runs so far, 0 failures, over 1m25s
33 runs so far, 0 failures, over 1m30s
46 runs so far, 0 failures, over 1m35s
48 runs so far, 0 failures, over 1m40s
48 runs so far, 0 failures, over 1m45s
48 runs so far, 0 failures, over 1m50s
48 runs so far, 0 failures, over 1m55s
49 runs so far, 0 failures, over 2m0s
56 runs so far, 0 failures, over 2m5s
63 runs so far, 0 failures, over 2m10s
64 runs so far, 0 failures, over 2m15s
64 runs so far, 0 failures, over 2m20s
64 runs so far, 0 failures, over 2m25s
65 runs so far, 0 failures, over 2m30s
68 runs so far, 0 failures, over 2m35s
80 runs so far, 0 failures, over 2m40s
80 runs so far, 0 failures, over 2m45s
80 runs so far, 0 failures, over 2m50s
81 runs so far, 0 failures, over 2m55s
82 runs so far, 0 failures, over 3m0s
84 runs so far, 0 failures, over 3m5s
94 runs so far, 0 failures, over 3m10s
96 runs so far, 0 failures, over 3m15s
97 runs so far, 0 failures, over 3m20s
98 runs so far, 0 failures, over 3m25s
98 runs so far, 0 failures, over 3m30s
100 runs so far, 0 failures, over 3m35s
111 runs so far, 0 failures, over 3m40s
112 runs so far, 0 failures, over 3m45s
113 runs so far, 0 failures, over 3m50s
114 runs so far, 0 failures, over 3m55s
115 runs so far, 0 failures, over 4m0s
116 runs so far, 0 failures, over 4m5s
125 runs so far, 0 failures, over 4m10s
129 runs so far, 0 failures, over 4m15s
130 runs so far, 0 failures, over 4m20s
131 runs so far, 0 failures, over 4m25s
131 runs so far, 0 failures, over 4m30s
134 runs so far, 0 failures, over 4m35s
140 runs so far, 0 failures, over 4m40s
145 runs so far, 0 failures, over 4m45s
146 runs so far, 0 failures, over 4m50s
147 runs so far, 0 failures, over 4m55s
148 runs so far, 0 failures, over 5m0s
155 runs so far, 0 failures, over 5m5s
160 runs so far, 0 failures, over 5m10s
161 runs so far, 0 failures, over 5m15s
162 runs so far, 0 failures, over 5m20s
163 runs so far, 0 failures, over 5m25s
165 runs so far, 0 failures, over 5m30s
170 runs so far, 0 failures, over 5m35s
177 runs so far, 0 failures, over 5m40s
178 runs so far, 0 failures, over 5m45s
179 runs so far, 0 failures, over 5m50s
181 runs so far, 0 failures, over 5m55s
182 runs so far, 0 failures, over 6m0s
186 runs so far, 0 failures, over 6m5s
192 runs so far, 0 failures, over 6m10s
194 runs so far, 0 failures, over 6m15s
196 runs so far, 0 failures, over 6m20s
197 runs so far, 0 failures, over 6m25s
201 runs so far, 0 failures, over 6m30s
202 runs so far, 0 failures, over 6m35s
210 runs so far, 0 failures, over 6m40s
211 runs so far, 0 failures, over 6m45s
212 runs so far, 0 failures, over 6m50s
213 runs so far, 0 failures, over 6m55s
217 runs so far, 0 failures, over 7m0s
223 runs so far, 0 failures, over 7m5s
227 runs so far, 0 failures, over 7m10s
228 runs so far, 0 failures, over 7m15s
229 runs so far, 0 failures, over 7m20s
232 runs so far, 0 failures, over 7m25s
234 runs so far, 0 failures, over 7m30s
240 runs so far, 0 failures, over 7m35s
244 runs so far, 0 failures, over 7m40s
244 runs so far, 0 failures, over 7m45s
245 runs so far, 0 failures, over 7m50s
248 runs so far, 0 failures, over 7m55s
251 runs so far, 0 failures, over 8m0s
256 runs so far, 0 failures, over 8m5s
260 runs so far, 0 failures, over 8m10s
261 runs so far, 0 failures, over 8m15s
261 runs so far, 0 failures, over 8m20s
265 runs so far, 0 failures, over 8m25s
269 runs so far, 0 failures, over 8m30s
274 runs so far, 0 failures, over 8m35s
276 runs so far, 0 failures, over 8m40s
277 runs so far, 0 failures, over 8m45s
279 runs so far, 0 failures, over 8m50s
282 runs so far, 0 failures, over 8m55s
286 runs so far, 0 failures, over 9m0s
291 runs so far, 0 failures, over 9m5s
292 runs so far, 0 failures, over 9m10s
294 runs so far, 0 failures, over 9m15s
297 runs so far, 0 failures, over 9m20s
299 runs so far, 0 failures, over 9m25s
302 runs so far, 0 failures, over 9m30s
308 runs so far, 0 failures, over 9m35s
309 runs so far, 0 failures, over 9m40s
311 runs so far, 0 failures, over 9m45s
315 runs so far, 0 failures, over 9m50s
316 runs so far, 0 failures, over 9m55s
320 runs so far, 0 failures, over 10m0s
325 runs so far, 0 failures, over 10m5s
326 runs so far, 0 failures, over 10m10s
329 runs so far, 0 failures, over 10m15s
332 runs so far, 0 failures, over 10m20s
333 runs so far, 0 failures, over 10m25s
338 runs so far, 0 failures, over 10m30s
341 runs so far, 0 failures, over 10m35s
342 runs so far, 0 failures, over 10m40s
347 runs so far, 0 failures, over 10m45s
348 runs so far, 0 failures, over 10m50s
350 runs so far, 0 failures, over 10m55s
356 runs so far, 0 failures, over 11m0s
358 runs so far, 0 failures, over 11m5s
360 runs so far, 0 failures, over 11m10s
364 runs so far, 0 failures, over 11m15s
365 runs so far, 0 failures, over 11m20s
368 runs so far, 0 failures, over 11m25s
374 runs so far, 0 failures, over 11m30s
374 runs so far, 0 failures, over 11m35s
378 runs so far, 0 failures, over 11m40s
381 runs so far, 0 failures, over 11m45s
381 runs so far, 0 failures, over 11m50s
388 runs so far, 0 failures, over 11m55s
390 runs so far, 0 failures, over 12m0s
391 runs completed, 1 failures, over 12m3s
FAIL
```

Please assign, take a look and update the issue accordingly.

--- a/pkg/storage/node_liveness_test.go
+++ b/pkg/storage/node_liveness_test.go
@@ -59,7 +59,6 @@ func stopNodeLivenessHeartbeats(mtc *multiTestContext) {
 
 func TestNodeLiveness(t *testing.T) {
 	defer leaktest.AfterTest(t)()
-	t.Skip("#9973")
 	mtc := startMultiTestContext(t, 3)
 	defer mtc.Stop()
 
@@ -89,9 +88,9 @@ func TestNodeLiveness(t *testing.T) {
 	verifyLiveness(t, mtc)
 
 	// Verify metrics counts.
-	for _, nl := range mtc.nodeLivenesses {
-		if c := nl.Metrics().HeartbeatSuccesses.Count(); c != 2 {
-			t.Errorf("expected metrics count == 2; got %d", c)
+	for i, nl := range mtc.nodeLivenesses {
+		if c := nl.Metrics().HeartbeatSuccesses.Count(); c < 2 {
+			t.Errorf("node %d: expected metrics count >= 2; got %d", (i + 1), c)
 		}
 	}
 }

